query/src/pipelines/new/processors/transforms/hash_join/chaining_hash_table.rs

```diff
@@ -12,28 +12,42 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+use std::borrow::BorrowMut;
+use std::hash::Hash;
 use std::sync::Arc;
 use std::sync::Mutex;
-use std::sync::RwLock;
 
+use common_base::infallible::RwLock;
 use common_datablocks::DataBlock;
+use common_datablocks::HashMethod;
+use common_datablocks::HashMethodFixedKeys;
+use common_datablocks::HashMethodSerializer;
 use common_datavalues::Column;
 use common_datavalues::ColumnRef;
 use common_datavalues::ConstColumn;
 use common_datavalues::DataSchemaRef;
-use common_datavalues::Series;
-use common_datavalues::SmallVu8;
+use common_datavalues::PrimitiveType;
 use common_exception::Result;
 use common_planners::Expression;
 
 use crate::common::ExpressionEvaluator;
 use crate::common::HashMap;
+use crate::common::HashTableKeyable;
+use crate::pipelines::new::processors::transforms::hash_join::row::Chunk;
 use crate::pipelines::new::processors::transforms::hash_join::row::RowPtr;
 use crate::pipelines::new::processors::transforms::hash_join::row::RowSpace;
 use crate::pipelines::new::processors::HashJoinState;
 use crate::pipelines::transforms::group_by::keys_ref::KeysRef;
 use crate::sessions::QueryContext;
 
+pub enum HashTable {
```

Comment:
Maybe you can put the `HashMethod` here?

Reply:
Do you mean
```rust
pub enum HashTable {
    SerializerHashTable(HashMethodSerializer),
    KeyU8HashTable(HashMethodFixedKeys<u8>),
    ...
}
```

Reply:
> Do you mean
> 
> ```rust
> 
> pub enum HashTable {
> 
>     SerializerHashTable(HashMethodSerializer),
> 
>     KeyU8HashTable(HashMethodFixedKeys<u8>),
> 
>     ...
> 
> }
> 
> ```

Yes, it's better to associate the structs with same generic types together.